src/entity_counter_recalculator.py

```diff
@@ -0,0 +1,183 @@
+                                            aws_connection_string_location=AWS_CONNECTION_STRING_LOCATION)
+    conn = connect_to_postgres(connection_info=connection_info,connection_timeout=CONNECTION_TIMEOUT)
+    records = __run_sql(conn)
+
+    if ENABLE_CLOUDWATCH_INTEGRATION:
+        logger.info("setting cloudwatch metrics")
+        cloudwatch = get_aws_client(resource_type="cloudwatch", localstack=LOCALSTACK,
+                                    region=REGION, localstack_address=LOCALSTACK_ADDRESS)
+        set_cloudwatch_metrics(records, cloudwatch, connection_info)
+
+    conn.close()
+
+    return records
+
+
+def set_cloudwatch_metrics(records, cloudwatch, connection_info):
+    customer_delta = 0
+    customer_deletes_needed = 0
+    metric_data = []
+    dimensions = [{
+                    'Name': "TABLE_NAME",
+                    'Value': connection_info["database"]
+                },
+                {
+                    'Name': "APP_VERSION",
+                    'Value': APP_VERSION
+                },
+                {
+                    'Name': "RECALCULATOR_NAME",
+                    'Value': "entity_counter_recalculator"
+                }]
+
+
+    for key in records["customerImages"]:
+        logger.info(f"customer images delta found - {key}")
+        if key["delta"] is not None:
+            customer_delta += abs(key["delta"])
+        else:
+            customer_deletes_needed += 1
+
+    metric_data.extend([
+        {
+            'MetricName': CLOUDWATCH_CUSTOMER_DIFFERENCE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': customer_delta
+        },
+        {
+            'MetricName': CLOUDWATCH_CUSTOMER_DELETE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': customer_deletes_needed
+        }
+    ])
+
+    space_delta = 0
+    space_deletes_needed = 0
+
+    for key in records["spaceImages"]:
+        logger.info(f"space images delta found - {key}")
+        if key["delta"] is not None:
+            space_delta += abs(key["delta"])
+        else:
+            space_deletes_needed += 1
+
+    metric_data.extend([
+        {
+            'MetricName': CLOUDWATCH_SPACE_DIFFERENCE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': space_delta
+        },
+        {
+            'MetricName': CLOUDWATCH_SPACE_DELETE_METRIC_NAME,
+            'Dimensions': dimensions,
+            'Unit': 'None',
+            'Value': space_deletes_needed
+        }
+    ])
+
+    try:
+        logger.debug(f"updating cloudwatch metrics - {metric_data}")
+        cloudwatch.put_metric_data(MetricData=metric_data,
+                                   Namespace='Entity Counter Recalculator')
+        return metric_data
+    except Exception as e:
+        logger.error(f"Error posting to cloudwatch: {e}")
+        raise e
+
+
+def __run_sql(conn):
```

Comment:
Similar to customer-storage, could `space-images` be worked out then `customer-images` by suming those?